alerter.c

```diff
@@ -4,11 +4,22 @@
 int alertFailureCount = 0;
 
 int networkAlertStub(float celcius) {
-    printf("ALERT: Temperature is %.1f celcius.\n", celcius);
+    
+    int threshold = 200;
+    int alert = 0;
     // Return 200 for ok
     // Return 500 for not-ok
-    // stub always succeeds and returns 200
-    return 200;
+    if(celcius >= threshold)
+    {
+        printf("ALERT: Temperature is %.1f celcius.\n", celcius);
+        alert = 500;
+    }
+    else
+    {
+        alert = 200;
+    }
+
+    return alert;
 }
 
 void alertInCelcius(float farenheit) {

```

Comment:
The production code should not be aware of test code or test stub. It should be abstracted by the use of interface or function pointer.